components/brave_wallet/common/hash_utils.cc

```diff
@@ -75,4 +79,14 @@ Ripemd160HashArray Hash160(base::span<const uint8_t> input) {
   return result;
 }
 
+SHA512HashArray HmacSha512(base::span<const uint8_t> key,
+                           base::span<const uint8_t> data) {
+  SHA512HashArray hmac;
+  unsigned int out_len = 0;
+  CHECK(HMAC(EVP_sha512(), key.data(), key.size(), data.data(), data.size(),
+             hmac.data(), &out_len));
+  CHECK_EQ(out_len, hmac.size());
```

Comment:
same as above, this is not an appropriate use of `CHECK`. Any problem with the generation of the hmac should be handled by the caller, not by crashing. Maybe this should return `std::optional<SHA512HashArray>`?